### PR TITLE
adding secret to fetch truststore password

### PR DESCRIPTION
**Pull request by Shehanir**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: apim-operator/pkg/controller/api/api_controller.go [hunk 1]
```diff
@@ -1118,3 +1120,44 @@ func getAnalyticsPVClaim(r *ReconcileAPI, deployVolumeMount []corev1.VolumeMount
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      truststoreSecretName,
+				Namespace: wso2NameSpaceConst,
+			},
+		}
+		truststoresecret.Data = map[string][]byte{
+			truststoreSecretData: []byte(encodedpassword),
+		}
+		errsecret := r.client.Create(context.TODO(), truststoresecret)
+		log.Error(errsecret ,"error in creating trustsote password")
```
Comment (npamudika) [inline apim-operator/pkg/controller/api/api_controller.go]: Spelling mistake in truststore

File: apim-operator/pkg/controller/api/api_controller.go [hunk 2]
```diff
@@ -1118,3 +1120,44 @@ func getAnalyticsPVClaim(r *ReconcileAPI, deployVolumeMount []corev1.VolumeMount
...
+
+func getTruststorePassword(r *ReconcileAPI) string {
+
+	var password string;
+	//get secret if available
+	secret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: truststoreSecretName, Namespace: wso2NameSpaceConst},
+	secret)
+	if err != nil && errors.IsNotFound(err){
+		encodedpassword := encodedTrustsorePassword
```
Comment (renuka-fernando) [inline apim-operator/pkg/controller/api/api_controller.go]: Spelling mistake in `encodedTruststorePassword`

File: apim-operator/pkg/controller/api/api_controller.go (hunk 1 above)
Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go:1152]: Resolved

File: apim-operator/pkg/controller/api/api_controller.go (hunk 2 above)
Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go:1132]: Resolved

**Pull request merged**